editor/test/effects.js

```diff
@@ -321,6 +322,96 @@ describe( 'effects', () => {
 		} );
 	} );
 
+	describe( '.TRASH_POST', () => {
+		it( 'should dispatch an removal notice for trashing a post as well as a trash post success action.', () => {
+			const apiCall = ( post ) => ( {
+				getPostTypeModel: function( postType ) { // eslint-disable-line no-unused-vars
+					return function( postObject ) { // eslint-disable-line no-unused-vars
+						return {
+							destroy: function() {
+								return this;
+							},
+							then: function( callback, errorHandler ) {
+								if ( post !== 'error' ) {
+									callback();
+								} else {
+									errorHandler( 'error' );
+								}
+							},
+						};
+					};
+				},
+			} );
+			const post = {
+				id: 1,
+				title: {
+					raw: 'A History of Pork',
+				},
+				content: {
+					raw: '',
+				},
+				status: 'draft',
+			};
+			wp.api = apiCall( post );
+
+			const handler = effects.TRASH_POST;
+			const dispatch = jest.fn();
+			const store = { getState: () => ( { currentPost: post } ), dispatch };
+
+			const action = { postId: 1 };
+
+			handler( action, store );
+
+			expect( dispatch ).toHaveBeenCalledTimes( 2 );
+			expect( dispatch.mock.calls[ 0 ][ 0 ] ).toEqual( removeNotice( 'TRASH_POST_NOTICE_ID' ) );
```

Comment:
Is there by any chance a nicer API offered by Jest to verify if the method was called with proper values? `toHaveBeenCalledWith` could work, but it doesn't check order. Maybe that would enough though.

Reply:
I like checking the exact stuff. toHaveBeenCalledWith is fine, if the array lookup is undesirable. Will change.

Reply:
It exposes the internal structure of the mock object at the moment, which is hard to follow in my opinion.